src/types/cards.ts

```diff
@@ -0,0 +1,245 @@
+import {ActionConfig} from "./actions";
+
+export namespace cards {
+  /**
+   * Options property of the Abstract Card.
+   *
+   * @property {string} type The type of the card.
+   * @property {string} icon Icon of the card.
+   * @property {ActionConfig} [tap_action] Action to perform on tapping a card.
+   * @property {ActionConfig} [double_tap_action] Action to perform on double-tapping a card.
+   * @property {ActionConfig} [hold_action] Action to perform on holding a card.
+   */
+  export interface abstractOptions {
+    type: string;
+    icon: string;
+    tap_action?: ActionConfig;
+    double_tap_action?: ActionConfig;
+    hold_action?: ActionConfig;
+  }
+
+  /**
+   * Entity Card.
+   *
+   * @property {string | undefined} entity The id of an entity.
+   */
+  export interface entityCard extends abstractOptions {
+    entity: string | undefined;
+  }
+
+  /**
+   * A horizontal stack of cards.
+   *
+   * @property {string} type The stack type.
+   * @property {Object[]} cards The content of the stack.
+   */
+  export interface horizontalStack {
+    type: "horizontal-stack",
+    cards: { [p: string]: any }[],
+  }
```

Comment:
Create a Card type, then reuse in the rest of the code base. `{ [p: string]: any }` currently exists in 6 different places in the codebase.
```suggestion

  export type Card = { [p: string]: any };
  
  /**
   * A horizontal stack of cards.
   *
   * @property {string} type The stack type.
   * @property {Object[]} cards The content of the stack.
   */
  export interface horizontalStack {
    type: "horizontal-stack",
    cards: Card[],
  }
  
```

Reply:
`{ [p: string]: any }` is not exactly a type which is bound to some entity.
It represent an object with an [index signature](
https://www.typescriptlang.org/docs/handbook/2/objects.html#index-signatures).

Those are in the code to force some type checking.
At those places I don't exactly know what the value will be, other then it's an object/array with a string index with some value.
Once again set to `any` for compiling, but maybe they can be further specified or set to `unknown`.